sys/net/application_layer/nanocoap/nanocoap.c

```diff
@@ -508,8 +508,20 @@ ssize_t coap_handle_req(coap_pkt_t *pkt, uint8_t *resp_buf, unsigned resp_buf_le
     if (pkt->hdr->code == 0) {
         return coap_build_reply(pkt, COAP_CODE_EMPTY, resp_buf, resp_buf_len, 0);
     }
-    return coap_tree_handler(pkt, resp_buf, resp_buf_len, ctx,
-                             coap_resources, coap_resources_numof);
+
+    ssize_t retval =  coap_tree_handler(pkt, resp_buf, resp_buf_len, ctx,
+                                        coap_resources, coap_resources_numof);
+
+    if (retval == -ECANCELED) {
```

Comment:
Where do you get the `-ECANCELED` from?
As far as I seem, it's only generated by `nanocoap_server_build_separate()`, but that would then be a separate response that does not get generated by `coap_handle_req()`.

Reply:
Ah, I should also update the implementation of `coap_build_reply()`, not just the API doc :sweat_smile:

Reply:
Will `coap_build_reply_header()` also need an update?